subworkflows/local/reference_indexing.nf

```diff
@@ -17,20 +18,77 @@ workflow REFERENCE_INDEXING {
 
     if ( fasta.extension == 'csv' || fasta.extension == 'tsv' ) {
         // If input (multi-)reference sheet supplied
         REFERENCE_INDEXING_MULTI ( fasta )
         ch_reference_for_mapping = REFERENCE_INDEXING_MULTI.out.reference
+        ch_mitochondrion_header  = REFERENCE_INDEXING_MULTI.out.mitochondrion_header
+        ch_hapmap                = REFERENCE_INDEXING_MULTI.out.hapmap
+        ch_pmd_mask              = REFERENCE_INDEXING_MULTI.out.pmd_mask
+        ch_snp_capture_bed       = REFERENCE_INDEXING_MULTI.out.snp_capture_bed
+        ch_pileupcaller_snp      = REFERENCE_INDEXING_MULTI.out.pileupcaller_snp
+        ch_sexdeterrmine_bed     = REFERENCE_INDEXING_MULTI.out.sexdeterrmine_bed
+        ch_bedtools_feature      = REFERENCE_INDEXING_MULTI.out.bedtools_feature
         ch_versions = ch_versions.mix( REFERENCE_INDEXING_MULTI.out.versions )
     } else {
         // If input FASTA and/or indicies supplied
         REFERENCE_INDEXING_SINGLE ( fasta, fasta_fai, fasta_dict, fasta_mapperindexdir )
+        ch_mitochondrion_header  = REFERENCE_INDEXING_SINGLE.out.mitochondrion_header
+        ch_hapmap                = REFERENCE_INDEXING_SINGLE.out.hapmap
+        ch_pmd_mask              = REFERENCE_INDEXING_SINGLE.out.pmd_mask
+        ch_snp_capture_bed       = REFERENCE_INDEXING_SINGLE.out.snp_capture_bed
+        ch_pileupcaller_snp      = REFERENCE_INDEXING_SINGLE.out.pileupcaller_snp
+        ch_sexdeterrmine_bed     = REFERENCE_INDEXING_SINGLE.out.sexdeterrmine_bed
+        ch_bedtools_feature      = REFERENCE_INDEXING_SINGLE.out.bedtools_feature
         ch_reference_for_mapping = REFERENCE_INDEXING_SINGLE.out.reference
         ch_versions = ch_versions.mix( REFERENCE_INDEXING_SINGLE.out.versions )
     }
+
+    // Filter out input options that are not provided
+    ch_mitochondrion_header = ch_mitochondrion_header
+                    .filter{ it[1] != "" }
+
+    ch_hapmap = ch_hapmap
+                    .filter{ it[1] != "" }
+
+    ch_pmd_mask = ch_pmd_mask
+                    .filter{ it[1] != "" && it[2] != "" }
+
+    ch_capture_bed = ch_snp_capture_bed //optional
+                    .branch {
+                        meta, capture_bed ->
+                        input: capture_bed != ""
+                        skip: true
+                    }
+    ch_capture_bed_gunzip = ch_capture_bed.input //unzip
+                    .branch {
+                        meta, capture_bed ->
+                        forgunzip: capture_bed.extension == "gz"
+                        skip: true
+                    }
+    GUNZIP_SNPBED( ch_capture_bed_gunzip.forgunzip )
```

Comment:
Why not filter out non-present entries, as the other cases above, and then branch immediately after then for the gzip and non-gzip [but existing] files (rather than two branches)? 

Then mix the gzipped and non-gzipped after rather than the two skips?

(Note I maybe missing something here)

Reply:
The bedfile input for qualimap is optional, so I need to preserve the non-present entries, too.

Reply:
Ah OK, rather than just dropping it entirely - OK!